tests/models/test_hooks.py

```diff
@@ -419,43 +460,25 @@ def teardown(self, stage=None):
         'on_validation_start',
         'on_epoch_start',
         'on_validation_epoch_start',
-        'on_validation_batch_start',
-        'on_before_batch_transfer',
-        'transfer_batch_to_device',
-        'on_after_batch_transfer',
-        'on_validation_batch_end',
+        *(model.val_batch * val_batches),
+        'validation_epoch_end',
         'on_validation_epoch_end',
         'on_epoch_end',
         'on_validation_end',
         'on_validation_model_train',
         'on_train_start',
         'on_epoch_start',
         'on_train_epoch_start',
-        'on_train_batch_start',
-        'on_before_batch_transfer',
-        'transfer_batch_to_device',
-        'on_after_batch_transfer',
-        'on_before_zero_grad',
-        'on_after_backward',
-        'on_train_batch_end',
-        'on_train_batch_start',
-        'on_before_batch_transfer',
-        'transfer_batch_to_device',
-        'on_after_batch_transfer',
-        'on_before_zero_grad',
-        'on_after_backward',
-        'on_train_batch_end',
+        *(model.train_batch * train_batches),
```

Comment:
the deleted lines do not match what is defined in model.train_batch
can you give a hint what that means?

Reply:
Opened https://github.com/PyTorchLightning/pytorch-lightning/pull/7689 to just move the class so it's easier to review